packages/mobile/src/navigator/Navigator.tsx

```diff
@@ -442,12 +443,12 @@ const settingsScreens = (Navigator: typeof Stack) => (
     <Navigator.Screen
       name={Screens.LinkBankAccountScreen}
       component={LinkBankAccountScreen}
-      options={headerWithBackButton}
+      options={headerWithBackButtonEvent({ eventName: CICOEvents.link_bank_account_cancel })}
```

Comment:
Between iOS and Android, there are 3 ways to leave a screen:
- hitting the back button in the navigator bar
- using the system back button (Android only)
- swipe to navigate back (iOS only)

This event will only catch the back button in the navigation bar.
One way to cover all cases is by using the technique described in https://reactnavigation.org/docs/preventing-going-back/
But we'd need to be sure it doesn't trigger when not intended. See when `beforeRemove` is actually called.

Reply:
@jeanregisser Interesting -- thanks for the info... I'm not sure I understand why it would trigger unintentionally; from the docs, it looks like the only case where we wouldn't want to fire an event is when `Some action such as pop or reset was dispatched which removes the screen from the state`. Is something like this possible? It seems like we would just need to add a listener for `beforeRemove` events and log an event there..

Reply:
Yeah if there's no `pop` or `reset` involved in this flow it should be good.